GoogleChromeLabs/ps-analysis-tool
# Refactor: Settings page and single tab processing feature

## Description
This PR includes some refactoring to the single-tab-processing feature, fixes to some issues which were introduced there and adds some relevant test coverage.

## Relevant Technical Choices
- Used `chrome.devtools.inspectedWindow.eval` instead of `chrome.tabs.query` to get the site URL.
- Added a loader to the DevTools which fixes the flash of the Analyze the tab screen.

## Testing Instructions
- Open the terminal and run `npm I && npm run dev`.
- Open any site and go to Privacy Sandbox in DevTools you should first see a loader for a brief period of time then see the cookies landing page.
- The problem of `First Party cookies not found should not persist for long anymore.

---
